testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/UmaRepresentationTest.java

```diff
@@ -0,0 +1,173 @@
+
+        RealmResource realm = getRealm();
+        String resourceServerId = getClient(realm).toRepresentation().getId();
+        UserRepresentation user = realm.users().search("kolo").get(0);
+
+        PolicyEvaluationRequest request = new PolicyEvaluationRequest();
+        request.setUserId(user.getId());
+        request.setClientId(resourceServerId);
+        request.addResource("Resource A", "ScopeA");
+        PolicyEvaluationResponse result = getClient(realm).authorization().policies().evaluate(request);
+        Assert.assertEquals(result.getStatus(), DecisionEffect.PERMIT);
+
+        List<PolicyEvaluationResponse.EvaluationResultRepresentation> evaluations = result.getResults();
+        Assert.assertFalse(evaluations.isEmpty());
+        Assert.assertEquals(1, evaluations.size());
+
+        List<PolicyEvaluationResponse.PolicyResultRepresentation> policies = evaluations.get(0).getPolicies();
+        Assert.assertFalse(evaluations.isEmpty());
+        Assert.assertEquals(1, evaluations.size());
+
+        String description = policies.get(0).getPolicy().getDescription();
+        Assert.assertTrue(description.startsWith("Resource owner (marta) grants access"));
+    }
+
+    @Test
+    public void testCanRepresentPolicyResultGrantOfResourceOwnedByClient() throws Exception {
+        resource = addResource("Resource A", getClient(getRealm()).toRepresentation().getId(), true, "ScopeA");
+        permission = getAuthzClient().protection().permission();
+        createPermissionTicket();
+
+        RealmResource realm = getRealm();
+        String resourceServerId = getClient(realm).toRepresentation().getId();
+        UserRepresentation user = realm.users().search("kolo").get(0);
+
+        PolicyEvaluationRequest request = new PolicyEvaluationRequest();
+        request.setUserId(user.getId());
+        request.setClientId(resourceServerId);
+        request.addResource("Resource A", "ScopeA");
+        PolicyEvaluationResponse result = getClient(realm).authorization().policies().evaluate(request);
+        Assert.assertEquals(result.getStatus(), DecisionEffect.PERMIT);
+
+        List<PolicyEvaluationResponse.EvaluationResultRepresentation> evaluations = result.getResults();
+        Assert.assertFalse(evaluations.isEmpty());
+        Assert.assertEquals(1, evaluations.size());
+
+        List<PolicyEvaluationResponse.PolicyResultRepresentation> policies = evaluations.get(0).getPolicies();
+        Assert.assertFalse(evaluations.isEmpty());
+        Assert.assertEquals(1, evaluations.size());
+
+        String description = policies.get(0).getPolicy().getDescription();
+        Assert.assertTrue(description.startsWith("Resource owner (resource-server-test) grants access"));
+    }
+
+    @Test
+    public void testCanRepresentResourceBeanOfResourceOwnedByUser() throws Exception {
+        resource = addResource("Resource A", "marta", true, "ScopeA");
+
+        testingClient.server().run(session -> {
```

Comment:
Tests should pass if you use a method reference here as follows https://github.com/keycloak/keycloak/blob/419d9c63510a95498e98807fc8e85758a14dc4ec/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AuthzCleanupTest.java#L102.

Reply:
I was stuck on how to pass the resource ID, since the reference has to be a static method. But finding the resource by name instead should work fine.